osdk/tests/examples_in_book/write_a_kernel_in_100_lines_templates/lib.rs

```diff
@@ -44,42 +45,41 @@ fn create_user_space(program: &[u8]) -> UserSpace {
         segment.write_bytes(0, program).unwrap();
         segment
     };
-    let user_address_space = {
-        const MAP_ADDR: Vaddr = 0x0040_0000; // The map addr for statically-linked executable
-
-        // The page table of the user space can be
-        // created and manipulated safely through
-        // the `VmSpace` abstraction.
-        let vm_space = VmSpace::new();
-        let mut cursor = vm_space.cursor_mut(&(MAP_ADDR..MAP_ADDR + nbytes)).unwrap();
-        let map_prop = PageProperty::new(PageFlags::RWX, CachePolicy::Writeback);
-        for frame in user_pages {
-            cursor.map(frame.into(), map_prop);
-        }
-        drop(cursor);
-        Arc::new(vm_space)
-    };
-    let user_cpu_state = {
-        const ENTRY_POINT: Vaddr = 0x0040_1000; // The entry point for statically-linked executable
-
-        // The user-space CPU states can be initialized
-        // to arbitrary values via the UserContext
-        // abstraction.
-        let mut user_cpu_state = UserContext::default();
-        user_cpu_state.set_rip(ENTRY_POINT);
-        user_cpu_state
-    };
-    UserSpace::new(user_address_space, user_cpu_state)
+
+    const MAP_ADDR: Vaddr = 0x0040_0000; // The map addr for statically-linked executable
+
+    // The page table of the user space can be
+    // created and manipulated safely through
+    // the `VmSpace` abstraction.
+    let vm_space = VmSpace::new();
+    let mut cursor = vm_space.cursor_mut(&(MAP_ADDR..MAP_ADDR + nbytes)).unwrap();
+    let map_prop = PageProperty::new(PageFlags::RWX, CachePolicy::Writeback);
+    for frame in user_pages {
+        cursor.map(frame.into(), map_prop);
+    }
+    drop(cursor);
+    vm_space
+}
+
+fn create_user_context() -> UserContext {
```

Comment:
This entire file is written in a top-down approach. The hierarchy of these functions is as follows.

* `main`
    * `create_vm_space`
        * `create_user_context`
    * `create_user_task`
        * `create_user_context`
        * `handle_syscall`

So this `create_user_context` function should be put _after_ `create_user_task`, not _before_ it.